### PR TITLE
Do not put temporary TF1 into ROOT's global function pool

### PR DESCRIPTION
**Pull request by Dr15Jones**

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoVertex/BeamSpotProducer/src/PVFitter.cc [hunk 1]
```diff
@@ -350,9 +350,10 @@ bool PVFitter::runFitter() {
     TH1F *h1PVz = (TH1F*) hPVx->ProjectionY("h1PVz", 0, -1, "e");
 
     //Use our own copy for thread safety
-    TF1 gausx("localGausX","gaus");
-    TF1 gausy("localGausY","gaus");
-    TF1 gausz("localGausZ","gaus");
+    // also do not add to global list of functions
+    TF1 gausx("localGausX","gaus",false);
```
Comment (pcanal) [inline RecoVertex/BeamSpotProducer/src/PVFitter.cc]: @Dr15Jones The 3rd argument in the case of TF1 is a different type than in the case of a TFormula.  For TF1, it is a TF1::EAddToList
```
   // Add to list behavior
   enum class EAddToList {
      kDefault,
      kAdd,
      kNo
   };
```
so this code actually 'still' request the addition to the list (i.e. kDefault)

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
Ping
This code caused a crash again in the integration build.

**Comment by pcanal** [comment]
I'll need a reproducer, the stack traces did not yield any good guesses on the cause.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
I found one more case of a TF1 being added to the global list. That was the last case in the package.

**Comment by Dr15Jones** [comment]
This change is purely technical. I was forced to also add the extra arguments to the TF1 constructor which match the defaults used by ROOT:
https://root.cern.ch/doc/master/classTF1.html#a460b83312445ffa2877255767fe99b21

**Comment by Dr15Jones** [comment]
type bug

**Comment by Dr15Jones** [comment]
urgent

**Comment by perrotta** [comment]
+1
I am not able to reproduce the crashes (same wfs as in #18620, slc6_amd64_gcc530, etc...)
Do not adding temporary TF1's to the root global list makes sense, however. All other defaults are maintained.
I trust @Dr15Jones for effectiveness of the bug-fix.

**Comment by Dr15Jones** [comment]
To be more clear, I do not promise that this will fix the crash we are seeing. By getting this in, if we have another crash we will have eliminated one of the most obvious concentenders we presently have. Plus, not putting the TF1 into the global list is just a plain good idea to avoid other potential problems as well as make the code a bit more thread efficient.

**Pull request merged**